### PR TITLE
[TF] Remove older Tensorflow versions support

### PR DESCRIPTION
**Pull request by nikita-savelyevv**
### Changes

Drop support for TF 2.4-2.7

### Reason for changes

These versions are not tested anymore.

### Related tickets

108878

### CODE REVIEW

File: nncf/experimental/tensorflow/patch_tf.py [hunk 1]
```diff
@@ -198,12 +197,6 @@ def patch_tf_operations() -> None:
                         curr_module = getattr(curr_module, curr_name)
                     setattr(curr_module, name, tf_op_wrapper)
 
-            # TODO(andrey-churkin): Changes references from the `tensorflow.python.ops.nn`
```
Comment (nikita-savelyevv) [inline nncf/experimental/tensorflow/patch_tf.py:201]: @andrey-churkin as I understand this is save to delete now?

**Comment by nikita-savelyevv** [comment]
run pytorch pre-commit tests
run tensorflow pre-commit tests
run openvino pre-commit tests
run onnx pre-commit tests

**Review by alexsu52 — COMMENTED** [review]
Could you provide the sota test results for this PR?

File: nncf/experimental/tensorflow/patch_tf.py (hunk 1 above)
Comment (andrey-churkin) [reply nncf/experimental/tensorflow/patch_tf.py:201]: Yes

**Review by alexsu52 — APPROVED** [review]

**Pull request merged**